### PR TITLE
Implemented Issue #12 -- show unbound phases & lifecycle

### PR DESCRIPTION
**Pull request by malice00**
- Added new parameter 'buildplan.showLifecycles' for goals 'list' and 'list-phase'; When this parameter is set to 'true', the output will show in which lifecycle a phase was defined (if any).
- Added new parameter 'buildplan.showAllPhases' for goal 'list-phase'; This parameter will print all phases, even if no mapping to an execution is available
- Added several tests for the new parameters
- Added documentation
- Necessary changes to other (test-)classes for handling of lifecycle

### CODE REVIEW

**Comment by jcgay** [comment]
Wow thanks ! I will have a look into it 👍

**Comment by malice00** [comment]
Please do. If something needs a little work, let me know -- or fix it yourself, whatever is easiest for you!

File: pom.xml [hunk 1]
```diff
@@ -185,6 +196,7 @@
                     <cloneClean>true</cloneClean>
                     <showVersion>true</showVersion>
                     <postBuildHookScript>verify.groovy</postBuildHookScript>
+                    <mavenOpts>${argLine}</mavenOpts>
```
Comment (jcgay) [inline pom.xml:199]: Oh great to add ITs in code coverage 👍

**Review by jcgay — APPROVED** [review]
This is really great ! Thx for spending times on this 👍 
It seems fine to me, I merge and will publish a new release soon :)

**Comment by jcgay** [comment]
I have finally find time to publish a new release (v1.5). Thanks again ! 👍

**Pull request closed without merging**